src/storage/redis_db.cc

```diff
@@ -558,10 +611,24 @@ rocksdb::Status SubKeyScanner::Scan(RedisType type, const Slice &user_key, const
   uint64_t cnt = 0;
   std::string ns_key = AppendNamespacePrefix(user_key);
   Metadata metadata(type, false);
+  std::string raw_value;
+  Slice rest;
   LatestSnapShot ss(storage_);
-  rocksdb::Status s = GetMetadata(GetOptions{ss.GetSnapShot()}, {type}, ns_key, &metadata);
+  rocksdb::Status s = GetMetadata(GetOptions{ss.GetSnapShot()}, {type}, ns_key, &raw_value, &metadata, &rest);
   if (!s.ok()) return s;
 
+  // for hash type, we should filter expired field if encoding is with_ttl
+  bool is_encoding_field_ttl = false;
+  if (metadata.Type() == kRedisHash && !rest.empty()) {
+    uint8_t field_encoding = 0;
+    if (!GetFixed8(&rest, &field_encoding)) {
+      return rocksdb::Status::InvalidArgument();
+    }
+    if (field_encoding == 1) {
```

Comment:
Please use `HashSubkeyType` instead of hard-coding here.

Reply:
Thank for you point it up. I had fixed it.